File: shop/src/main/java/com/shopping/shop/entity/Product.java
package com.shopping.shop.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "shop_product")
@Getter
@Setter
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    private String code;

    private double price;

    private String imageURL;

    private String description;

}

File: shop-front/shop_front/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String code;
  final String imageURL;
  final double price;

  Product({required this.id,required this.name, required this.code, required this.imageURL, required this.price});

  factory Product.fromJson(Map<String, dynamic> json){
    return Product(
      id: json['id'],
      name: json['name'],
      code: json['code'],
      imageURL: json['imageURL'],
      price: json['price']
    );
  }

}


